py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -316,31 +315,27 @@ def _request(self, method, url, body=None):
                 response = http.request(method, url, body=body, headers=headers)
             statuscode = response.status
         data = response.data.decode("UTF-8")
-        LOGGER.debug(f"Remote response: status={response.status} | data={data} | headers={response.headers}")
+        LOGGER.debug(f"Remote response: data={data} | headers={response.headers}")
         try:
             if 300 <= statuscode < 304:
                 return self._request("GET", response.headers.get("location", None))
```

Comment:
this code looks suspicious to me as it might potentially fall into an endless recursion. Usually clients (including urllib3, which is this code base on) handle such stuff automatically with some default number of allowed redirect retries to avoid the above recursion issue.

Reply:
I was actually testing this out earlier this week for Sauce-related reasons... The default number of allowed redirects is 3, and that is actually what happens. You can set remote URL to http://demo.cyotek.com/features/redirectlooptest.php and see that you only get 3. I'm not sure what this loop does, I just know it didn't result in infinite retries.